governance/src/test/java/org/apache/servicecomb/governance/FlowControlTest.java

```diff
@@ -123,4 +132,93 @@ public void test_rate_limiting_service_name_work() throws Throwable {
     Assertions.assertTrue(expected.get());
     Assertions.assertFalse(notExpected.get());
   }
+
+  @Test
+  public void test_delay_fault_injection_service_name_work() throws Throwable {
+    DecorateCheckedSupplier<Object> ds = Decorators.ofCheckedSupplier(() -> "test");
+
+    GovernanceRequest request = new GovernanceRequest();
+    request.setUri("/faultInjectDelay");
+    request.setServiceName("srcService");
+
+    Fault fault = faultInjectionHandler.getActuator(request);
+
+    Assertions.assertEquals("test", ds.get());
+
+    // flow control
+    CountDownLatch cd = new CountDownLatch(10);
+    AtomicBoolean notExpected = new AtomicBoolean(false);
+    AtomicBoolean delayExpected = new AtomicBoolean(false);
+    AtomicBoolean abortExpected = new AtomicBoolean(false);
+    for (int i = 0; i < 10; i++) {
+      new Thread(() -> {
+        fault.injectFault(faultResponse -> {
+          if (faultResponse.isDelay()) {
+            delayExpected.set(true);
+          } else if (!faultResponse.isSuccess()) {
+            abortExpected.set(true);
+          } else {
+            try {
+              Object result = ds.get();
+              if (!"test".equals(result)) {
+                notExpected.set(true);
+              }
+            } catch (Throwable e) {
+              notExpected.set(true);
+            }
+          }
+          cd.countDown();
+        });
+      }).start();
+    }
+    //timeout should be bigger than delayTime
+    cd.await(10, TimeUnit.SECONDS);
+    Assertions.assertFalse(notExpected.get());
+    Assertions.assertFalse(abortExpected.get());
+    Assertions.assertTrue(delayExpected.get());
+  }
+
+  @Test
+  public void test_abort_fault_injection_service_name_work() throws Throwable {
+    DecorateCheckedSupplier<Object> ds = Decorators.ofCheckedSupplier(() -> "test");
+
+    GovernanceRequest request = new GovernanceRequest();
+    request.setUri("/faultInjectAbort");
+    request.setServiceName("srcService");
+
+    Fault fault = faultInjectionHandler.getActuator(request);
+
+    Assertions.assertEquals("test", ds.get());
+
+    // flow control
+    CountDownLatch cd = new CountDownLatch(10);
+    AtomicBoolean notExpected = new AtomicBoolean(false);
+    AtomicBoolean delayExpected = new AtomicBoolean(false);
+    AtomicBoolean abortExpected = new AtomicBoolean(false);
+    for (int i = 0; i < 10; i++) {
+      new Thread(() -> {
+        fault.injectFault(faultResponse -> {
```

Comment:
From the usage, seems the users need to call Fault api himself. This is not a correct usage case. 

see FlowControlTests for usage exampes. We should call ds.get and expecting errors.